sdk/src/main/scala/com/horizen/api/http/SidechainWalletApiRoute.scala

```diff
@@ -134,6 +138,122 @@ case class SidechainWalletApiRoute(override val settings: RESTApiSettings,
     }
   }
 
+  /**
+   * Import a private key inside the wallet
+   */
+  def importSecret: Route = (post & path("importSecret")) {
+    withAuth {
+      entity(as[ReqImportSecret]) { body =>
+        val secret = sidechainSecretsCompanion.parseBytes(BytesUtils.fromHexString(body.privKey))
+        val future = sidechainNodeViewHolderRef ? LocallyGeneratedSecret(secret)
+        Await.result(future, timeout.duration).asInstanceOf[Try[Unit]] match {
+          case Success(_) =>
+            ApiResponseUtil.toResponse(RespCreatePrivateKey25519(secret.publicImage()))
+          case Failure(e) =>
+            ApiResponseUtil.toResponse(ErrorSecretAlreadyPresent("Failed to add the key.", JOptional.of(e)))
+        }
+      }
+    }
+  }
+
+  /**
+   * Export a private key from the wallet based on its public key
+   */
+  def exportSecret: Route = (post & path("exportSecret")) {
+    withAuth {
+      entity(as[ReqExportSecret]) { body =>
+        val proposition = PublicKey25519PropositionSerializer.getSerializer.parseBytes(BytesUtils.fromHexString(body.publickey))
```

Comment:
User may request an export of VRF key or any custom key for example.
We need to consider:
1) new proposition <-> secret interface defined by Paolo C.
2) the case of "complex" proposition that may be related to multiple secrets. (Maybe not a case for this method, since it is all about the keys only)

A new wallet method that takes public key bytes and compares with  `secret.publicImage().publicKeyBytes()` can help.

Reply:
agree. all these  endpoints implementations are only related to 25519 keys
we could maybe change the endpoints path to something like:
wallet/secrets/25519/import
wallet/secrets/25519/export
wallet/secrets/25519/importList
wallet/secrets/25519/exportList

or add an additional  "type" paremeter to every endpoint , and just raise an exception if type != 25519

otherwise we have to change the implementation to be more generic: for the generic  proposition->secret there is this new method I added in wallet (note that 1 proposition could be unlocked by more than 1 secret)
< S extends Secret> List< S > secretsByProposition(ProofOfKnowledgeProposition< S > proposition)

Reply:
Done